# RDS
---

Amazon RDS for PostgreSQL es un servicio de base de datos gestionado que facilita la configuración, operación y escalabilidad de bases de datos PostgreSQL en la nube. Proporciona almacenamiento de datos histórico con opciones de alta disponibilidad y escalabilidad automática, lo que lo convierte en una solución ideal para grandes volúmenes de datos. Multi-AZ ofrece redundancia y tolerancia a fallos, asegurando que los datos siempre estén disponibles, incluso en caso de fallos en un nodo.

In [1]:
import json

# Cargar el archivo JSON
file_path_json = 'My Estimate.json'

# Leer el archivo JSON
with open(file_path_json, encoding='latin-1') as f:
    data = json.load(f)

# Buscar el servicio Amazon RDS for PostgreSQL
for service in data['Grupos']['Servicios']:
    if service['Nombre del servicio'].strip() == 'Amazon RDS for PostgreSQL':
        rds_service = service
        break

# Imprimir los detalles del servicio Amazon RDS for PostgreSQL
print("Nombre del servicio:", rds_service['Nombre del servicio'])
print("Descripción:", rds_service['Descripción'])
print("Región:", rds_service['Región'])
print("Costo mensual:", rds_service['Costo del servicio']['mensual'])
print("Costo total (12 meses):", rds_service['Costo del servicio']['12 meses'])
print("Propiedades:")
for key, value in rds_service['Propiedades'].items():
    print(f"  {key}: {value}")


Nombre del servicio: Amazon RDS for PostgreSQL
Descripción: Almacenamiento histórico
Región: Este de EE. UU. (Ohio)
Costo mensual: 3141.20
Costo total (12 meses): 37694.40
Propiedades:
  Cantidad de almacenamiento: 100 GB
  Volumen de almacenamiento: SSD de uso general (gp2)
  Nodos: 2
  Tipo de instancia: db.r5.2xlarge
  Utilización (solo bajo demanda): 100 %Utilized/Month
  Opción de implementación: Multi-AZ
  Modelo de precios: OnDemand
  Almacenamiento de copias de seguridad adicional: 0 GB


RDS se ha configurado para gestionar el almacenamiento de datos históricos de consumo energético. La opción seleccionada es PostgreSQL, que proporciona un equilibrio entre rendimiento y capacidad de manejo de consultas complejas. Se ha configurado con una implementación Multi-AZ para garantizar la alta disponibilidad y la tolerancia a fallos, lo que asegura la continuidad del servicio incluso en casos de interrupciones.

El almacenamiento asignado es de 100 GB, suficiente para manejar los datos históricos de los clientes. El tipo de almacenamiento es SSD de uso general (gp2), que garantiza un rendimiento sólido para cargas de trabajo de lectura y escritura moderadas. El tipo de instancia seleccionado es db.r5.2xlarge, que ofrece un alto rendimiento con 8 vCPUs y 64 GB de RAM, asegurando que la base de datos pueda manejar eficientemente las consultas pesadas de datos.

El modelo de precios es OnDemand, que permite flexibilidad en el uso sin compromisos de largo plazo. Esta configuración asegura que la empresa pueda ajustar sus recursos de base de datos según la demanda, sin pagar por capacidad que no se utiliza.

El costo mensual para esta configuración es de  $3,141.20 USD.

El costo total de $37,694.40 USD para un período de 12 meses.


## Creación

```bash
aws rds create-db-instance `
    --db-instance-identifier Grupo4-mysql `
    --db-instance-class db.t4g.micro `
    --engine mysql `
    --allocated-storage 20 `
    --master-username "unisabanagrupo4" `
    --master-user-password "unisabanagrupo4" `
    --backup-retention-period 0 `
    --vpc-security-group-ids "sg-0c2557b5f8de630dd" `
    --db-name "grupo4_db" `
    --port 3306 `
```

Configuración RDS en AWS

![Configuración RDS en AWS](RDS.JPG)